examples/messari-graphclient/src/resolvers-backup.ts

```diff
@@ -0,0 +1,89 @@
+              protocolName,
+            },
+            info,
+          }).then((liquidityPools) =>
+            // We send protocolName here so we can take it in the resolver above
+            liquidityPools.map((liquidityPool) => {
+              const dailySnapshots = liquidityPool.dailySnapshots.map((snapshot) => {
+                //////////////////////
+
+                //ADDING FeesUSD:
+                const FeesUSD_result =
+                  snapshot.pool.fees !== undefined && snapshot.pool.fees.length > 0
+                    ? snapshot.dailyVolumeUSD *
+                      snapshot.pool.fees.filter((tf) => tf.feeType == 'FIXED_TRADING_FEE')[0].feePercentage
+                    : undefined
+
+                //ADDING TokenFeesUSD:
+                const TokenFeesUSD_result =
+                  snapshot.pool.fees !== undefined &&
+                  snapshot.pool.fees.length > 0 &&
+                  snapshot.dailyVolumeByTokenUSD.length > 0
+                    ? snapshot.dailyVolumeByTokenUSD.map((i) => {
+                        return i * snapshot.pool.fees.filter((tf) => tf.feeType == 'FIXED_TRADING_FEE')[0].feePercentage
+                      })
+                    : undefined
+
+                //ADDING TokenFeesRaw:
+                const TokenFeesRaw_result =
+                  snapshot.pool.fees !== undefined &&
+                  snapshot.pool.fees.length > 0 &&
+                  snapshot.dailyVolumeByTokenAmount.length > 0
+                    ? snapshot.dailyVolumeByTokenAmount.map((i) => {
+                        return i * snapshot.pool.fees.filter((tf) => tf.feeType == 'FIXED_TRADING_FEE')[0].feePercentage
+                      })
+                    : undefined
+
+                //ADDING RewardsPerToken:
+                const RewardsPerToken_result =
+                  snapshot.rewardTokenEmissionsUSD !== undefined && snapshot.rewardTokenEmissionsUSD.length > 0
+                    ? snapshot.rewardTokenEmissionsUSD.map((i) => {
+                        return i / snapshot.stakedOutputTokenAmount
+                      })
+                    : undefined
+
+                //ADDING RewardsROI:
+                //const RewardsROI_result =
+                //  snapshot.RewardsPerToken !== undefined && snapshot.RewardsPerToken.length > 0
+                //    ? snapshot.RewardsPerToken.map((i) => {
+                //        return i / snapshot.outputTokenPriceUSD
+                //      })
+                //    : undefined
+
+                return {
+                  ...snapshot,
+                  FeesUSD: FeesUSD_result,
+                  TokenFeesUSD: TokenFeesUSD_result,
+                  TokenFeesRaw: TokenFeesRaw_result,
+                  RewardsPerToken: RewardsPerToken_result,
+                  //RewardsROI: RewardsROI_result,
+                }
+              })
+
+              return {
+                ...liquidityPool,
+                protocolName,
+                dailySnapshots,
```

Comment:
If you use additional resolvers for the type of `snapshot`, you don't need any of this.

Reply:
Please check;
https://github.com/graphprotocol/graph-client/pull/142